src/main/java/org/terasology/minion/move/FindPathToNode.java

```diff
@@ -68,21 +71,26 @@ public void construct(final Actor actor) {
             moveComponent.path = Path.INVALID;
             return;
         }
-         pathfinderSystem.requestPath(
+        SettableFuture<List<Path>> pathFuture = pathfinderSystem.requestPath(
                 actor.getEntity(), currentBlock.getBlockPosition(),
                 Arrays.asList(workTarget.getBlockPosition()));
-                /*, new PathfinderSystem.PathReadyCallback() {
-                    @Override
-                    public void pathReady(int pathId, List<Path> path, WalkableBlock target, List<WalkableBlock> start) {
 
-                        if (path == null) {
+        Futures.addCallback(pathFuture, new FutureCallback<List<Path>>() {
+                    @Override
+                    public void onSuccess(List<Path> paths) {
+                        if (paths == null) {
                             moveComponent.path = Path.INVALID;
-                        } else if (path.size() > 0) {
-                            moveComponent.path = path.get(0);
+                        } else if (paths.size() > 0) {
+                            moveComponent.path = paths.get(0);
                         }
                         actor.save(moveComponent);
```

Comment:
Are you sure that this call is either thread-safe or always called on the main thread?

Reply:
I believe that it is? I changed the function to explicitly use a direct executor just to be save. This way, this call either runs directly in the action, or from pathfinding when the path is calculated, as found here: https://stackoverflow.com/questions/35211331/in-which-thread-futurecallback-will-be-called-for-a-listenablefuture-in-guava (this answer is for ListenableFuture instead of SettableFuture, but they are both implementations of Future so i'm assuming the executors work the same way)

Reply:
Ah okay, I get it now. I wondered why you even put it in a Future for the same thread if you could just call it directly. But the  `pathfinderSystem` seems to force you to use the `Future` approach. A bit odd, but ok.